src/main/java/org/reaktivity/nukleus/oauth/internal/stream/OAuthProxyFactory.java

```diff
@@ -674,12 +737,32 @@ private void onSignal(
         private void onTokenExpiredSignal(
             SignalFW signal)
         {
-            final long delay = grant.expiresAt - System.currentTimeMillis();
+            long delay = grant.expiresAt - System.currentTimeMillis();
```

Comment:
Extract `final long now = System.currentTimeMillis();` once and reuse for the duration of this method.